### ISSUE TITLE
Unable to npm run test on Node 4.4.7

### ISSUE THREAD

**Issue by amoshydra** [Hashnode/mern-starter#221]
I have applied a fix from #201 to get `npm run test` to run on my Windows.

However I am still getting a lot of errors stating 
`Error: proxies not supported on this platform. On v8/node/iojs, make sure to pass the --harmony_proxies flag`

Then I update my nodejs to v6.3.1 instead and it works.

~~So MERN actually isn't meant for node v4.x?~~
**Update:** Perhaps this is a problem on Windows only. `npm run test` works fine on my Vagrant/Ubuntu with Nodejs v4.4.7.

**Error stack**

```
> mern-starter@2.0.0 test C:\Projects\project-name
> cross-env NODE_ENV=test PORT=8080 MONGO_URL=mongodb://localhost:27017/mern-test node_modules/.bin/nyc node_modules/.bin/ava.cmd --harmony_proxies --serial

react-dom is an implicit dependency in order to support react@0.13-14. Please add the appropriate version to your devDependencies. See https://github.com/airbnb/enzyme#installation
C:\Projects\project-name\node_modules\enzyme\build\react-compat.js:111
      throw e;
      ^

Error: proxies not supported on this platform. On v8/node/iojs, make sure to pass the --harmony_proxies flag
    at new global.Proxy (C:\Projects\project-name\node_modules\harmony-reflect\reflect.js:2116:13)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\react\lib\SyntheticEvent.js:181:22)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\react\lib\SyntheticCompositionEvent.js:14:22)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\react\lib\BeforeInputEventPlugin.js:18:33)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\react\lib\ReactDefaultInjection.js:14:30)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\react\lib\ReactDOM.js:17:29)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\react-dom\index.js:3:18)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at C:\Projects\project-name\node_modules\enzyme\build\react-compat.js:107:18
    at Object.<anonymous> (C:\Projects\project-name\node_modules\enzyme\build\react-compat.js:187:5)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\enzyme\build\Utils.js:42:20)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\enzyme\build\MountedTraversal.js:31:14)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\enzyme\build\ReactWrapper.js:35:25)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\enzyme\build\index.js:6:21)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\babel-register\lib\node.js:156:7)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:16:3)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\client\modules\App\__tests__\App.spec.js:4:1)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at extensions.(anonymous function) (C:\Projects\project-name\node_modules\require-precompiled\index.js:13:11)
    at C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4
    at require.extensions.(anonymous function) (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:91:3)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\ava\lib\test-worker.js:95:1)
    at Module._compile (module.js:409:26)
    at Module.replacementCompile (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (C:\Projects\project-name\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\node_modules\append-transform\index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at Function.runMain (C:\Users\Amos Wong\.node-spawn-wrap-1884-dc0d95500043\node:40:10)
    at Object.<anonymous> (C:\Projects\project-name\node_modules\nyc\bin\wrap.js:19:4)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at Object.<anonymous> (C:\Users\Amos Wong\.node-spawn-wrap-1884-dc0d95500043\node:140:8)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:139:18)
    at node.js:968:3
react-dom is an implicit dependency in order to support react@0.13-14. Please add the appropriate version to your devDependencies. See https://github.com/airbnb/enzyme#installation
C:\Projects\project-name\node_modules\enzyme\build\react-compat.js:111
      throw e;
      ^

... more similar error ...


   16 passed
   7 exceptions


   × client\modules\App\__tests__\App.spec.js exited with a non-zero exit code: 1


   × client\modules\App\__tests__\Components\Footer.spec.js exited with a non-zero exit code: 1


   × client\modules\App\__tests__\Components\Header.spec.js exited with a non-zero exit code: 1


   × client\modules\Post\__tests__\components\PostCreateWidget.spec.js exited with a non-zero exit code: 1


   × client\modules\Post\__tests__\components\PostList.spec.js exited with a non-zero exit code: 1


   × client\modules\Post\__tests__\components\PostListItem.spec.js exited with a non-zero exit code: 1


   × server\models\__tests__\post.spec.js exited with a non-zero exit code: 1
--------------------------------------------------|----------|----------|----------|----------|----------------|
File                                              |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
--------------------------------------------------|----------|----------|----------|----------|----------------|
 client\modules\App\                              |      100 |      100 |      100 |      100 |                |
  AppActions.js                                   |      100 |      100 |      100 |      100 |                |
  AppReducer.js                                   |      100 |      100 |      100 |      100 |                |
 client\modules\App\components\                   |      100 |      100 |      100 |      100 |                |
  DevTools.js                                     |      100 |      100 |      100 |      100 |                |
 client\modules\Intl\                             |      100 |       90 |       50 |      100 |                |
  IntlActions.js                                  |      100 |      100 |      100 |      100 |                |
  IntlReducer.js                                  |      100 |       90 |      100 |      100 |                |
  IntlWrapper.js                                  |      100 |      100 |        0 |      100 |                |
 client\modules\Post\                             |      100 |      100 |    30.77 |      100 |                |
  PostActions.js                                  |      100 |      100 |       25 |      100 |                |
  PostReducer.js                                  |      100 |      100 |      100 |      100 |                |
 client\modules\Post\components\PostCreateWidget\ |        0 |        0 |        0 |        0 |                |
  PostCreateWidget.js                             |        0 |        0 |        0 |        0 |             12 |
 client\modules\Post\components\PostListItem\     |      100 |      100 |        0 |      100 |                |
  PostListItem.js                                 |      100 |      100 |        0 |      100 |                |
 client\util\                                     |      100 |    84.62 |      100 |      100 |                |
  apiCaller.js                                    |      100 |    84.62 |      100 |      100 |                |
 server\                                          |        0 |        0 |        0 |        0 |                |
  server.js                                       |        0 |        0 |        0 |        0 |... 116,144,145 |
--------------------------------------------------|----------|----------|----------|----------|----------------|
All files                                         |     62.5 |    47.06 |    33.33 |     62.5 |                |
--------------------------------------------------|----------|----------|----------|----------|----------------|


npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Amos Wong\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "test"
npm ERR! node v4.4.7
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE
npm ERR! mern-starter@2.0.0 test: `cross-env NODE_ENV=test PORT=8080 MONGO_URL=mongodb://localhost:27017/mern-test node_modules/.bin/nyc node_modules/.bin/ava.cmd --harmony_proxies --serial`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mern-starter@2.0.0 test script 'cross-env NODE_ENV=test PORT=8080 MONGO_URL=mongodb://localhost:27017/mern-test node_modules/.bin/nyc node_modules/.bin/ava.cmd --harmony_proxies --serial'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mern-starter package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     cross-env NODE_ENV=test PORT=8080 MONGO_URL=mongodb://localhost:27017/mern-test node_modules/.bin/nyc node_modules/.bin/ava.cmd --harmony_proxies --serial
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mern-starter
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mern-starter
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Projects\project-name\npm-debug.log
```

**Comment by fatman-** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;9fj1XHw0&#x2F;8-unable-to-run-npm-test-on-node-4-4-7"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Unable to run npm test on Node 4.4.7</a>

**Comment by somus** [comment]
Proxies are required for `babel-plugin-webpack-loaders` to work. I'm going to remove that dependency since it has more problems over its advantages. I'll release an updated version soon.

**Comment by putuyoga** [comment]
fyi this issue still happen (on my windows 10 machine) 😄

**Comment by mannyhenri** [comment]
closed due to duplicate and fixed with #198
